rust_src/src/vector_macros.rs

```diff
@@ -1,6 +1,6 @@
 macro_rules! offset_of {
     ($ty:ty, $field:ident) => {
-        &(*(0 as *const $ty)).$field as *const _ as usize
+        &(*(ptr::null() as *const $ty)).$field as *const _ as usize
```

Comment:
The `as *const $ty` should not be needed.

Reply:
I am open to another way to get it to compile. Just `ptr::null` does not work the compiler insists it does not know the type.

Reply:
Ah right. You could give a turbofish annotation. But that isn't more readable IMO.

Reply:
I tried `ptr::nul<$ty>()` and it did not like it.

Reply:
It would be `ptr::null::<$ty>()`. As I said, not particularly an improvement here.

Reply:
Yeah, not much.